File: backend/app/schemas/book.py
from pydantic import BaseModel


class Book(BaseModel):
    id: str
    title: str
    edition: str = ""
    category: str = ""
    authors: str = ""
    citations: int = 0
    editorialCount: int = 0
    editorialArea: str = ""

    year: str = ""
    editorial: str = ""
    doi: str = ""
    abstract: str = ""
    keywords: str = ""
    language: str = ""
    institution: str = ""
    matchMethod: str = ""
    openAlexId: str = ""
    editorialScore: float = 0.0
    citationScore: float = 0.0

File: frontend/app/types/book.ts
export interface Book {
  id: string;
  title: string;
  edition: string;
  category: string;
  authors: string;
  citations: number;
  editorialCount: number;
  editorialArea: string;

  year?: string;
  editorial?: string;
  doi?: string;
  abstract?: string;
  keywords?: string;
  language?: string;
  institution?: string;
  matchMethod?: string;
  openAlexId?: string;
  editorialScore?: number;
  citationScore?: number;
}